### ISSUE TITLE
Path resolves unexpectedly if the model is None

### ISSUE THREAD

**Issue by href** [morepath/morepath#500]
This seems to break certain things in my application:

```python
def test_absorb_error():

    import morepath

    class App(morepath.App):
        pass

    @App.path('/')
    class Root(object):
        pass

    @App.view(model=Root)
    def view_root(self, request):
        return request.path

    class File(object):
        def __init__(self, absorb):
            self.absorb = absorb

    @App.path('/files', model=File, absorb=True)
    def get_file(absorb):
        if absorb == 'foo':
            return File('foo')
        return None

    @App.view(model=File)
    def view_file(self, request):
        return request.path

    App.commit()

    client = Client(App())
    assert client.get('/?file=foo').text == '/'
    assert client.get('/files/foo').text == '/files/foo'

    # those checks fail as the 200: '/files/bar' is returned
    assert client.get('/files/bar').text != '/files/bar'
    assert client.get('/files/bar').status_code != 200
```

If the `/files` path returns None, the root view is called for some reason. This seems to be similar to #499. I unfortunately don't have time to get into this too much today, so for the release tomorrow I hope this test suffices.

I have some other things breaking that might be related to this, so I'll have to recheck those failures once this is fixed.

**Comment by faassen** [comment]
That's weird, I thought I fixed this very issue yesterday. I will look into it.

**Comment by faassen** [comment]
I reviewed matters and now fixed #499 and this issue better. This test passes now.
